src/prefect/cli/deploy.py

```diff
@@ -329,6 +329,14 @@ async def _run_single_deploy(
 
     deploy_config = await resolve_variables(deploy_config)
 
+    env_vars = dict(os.environ)
+    deploy_config["work_pool"]["name"] = apply_values(
+        deploy_config["work_pool"]["name"], env_vars
+    )
+    deploy_config["work_pool"]["work_queue_name"] = apply_values(
+        deploy_config["work_pool"]["work_queue_name"], env_vars
+    )
```

Comment:
Chose running `apply_values` on selected fields needed prior to the deployment as show in the diff, instead of iterating through all fields with something like
`deploy_config = apply_values(deploy_config, env_vars)`
because otherwise the following tests failed:

FAILED `tests/cli/test_deploy.py::TestProjectDeploySingleDeploymentYAML::test_project_deploy_templates_values - AssertionError: asse...`
`FAILED tests/cli/test_deploy.py::TestProjectDeploy::test_project_deploy_templates_values - AssertionError: assert None == 'foo'`

Reply:
@desertaxle would be curious if you have any thoughts here

Reply:
I think you can apply environment variables to the entire deployment config if you pass `remove_notset=True` to `apply_values`. That will ensure the placeholders are kept around for inserting the outputs from the `build` and `push` actions.

Reply:
thanks, actually, setting `remove_notset` to `False` did the trick!